subconscious/model.py

```diff
@@ -309,8 +297,24 @@ def get_index_key(cls, column_name):
                 first_iteration = False
             else:
                 result_set = result_set.intersection(temp_set)
-        futures = []
-        for key in sorted(result_set):
-            futures.append(cls.load(db, key))
+        return sorted(result_set)
 
-        return await asyncio.gather(*futures, loop=db.connection._loop)
+    @classmethod
+    async def filter_by(cls, db, **kwargs):
+        """Query by attributes iteratively. Ordering is not supported
+        Example:
+            User.get_by(db, age=[32, 54])
+            User.get_by(db, age=23, name="guido")
+
+        """
+        for key in await cls._get_ids_filter_by(db, **kwargs):
+            yield await cls.load(db, key)
+
+    @classmethod
+    async def get_object_or_none(cls, db, **kwargs):
```

Comment:
Django uses `get_object_or_None` (capital N):
https://github.com/skorokithakis/django-annoying#get_object_or_none-function

I'm fine with this way as well if you prefer, but it probably makes more sense to be consistent with what people are used to seeing